### DISCUSSION TITLE
Problem submitting Umami events from server

### DISCUSSION THREAD

**Discussion by geoffkendall** [umami-software/umami#2743] [Q&A]
Hi there,
I've got Umami tracking custom events from my client-side code (Javascript in a SvelteKit app), but when I try using the node library the events never appear.

Here's the code I'm using (note that I've hidden the websiteId in the code but it is configured correctly!

```
// Init
    umami.init({
      websiteId: "mywebsiteid", // Website id
      hostUrl: "https://track.tra.bz" // URL to your Umami instance
    })

    // Track - not working!
    const r = await umami.track("event", { data: { test: "test" } })
    console.log("Umami response: ", r)
```

The console response looks ok to me, but the event never gets logged:

```
Umami response: Response {
  status: 200,
  statusText: 'OK',
  headers: Headers {
    'access-control-allow-origin': '*',
    'cache-control': 'public, max-age=0, must-revalidate',
    connection: 'keep-alive',
    'content-length': '2',
    'content-security-policy': "default-src 'self';img-src *;script-src 'self' 'unsafe-eval' 'unsafe-inline';style-src 'self' 'unsafe-inline';connect-src 'self' api.umami.is cloud.umami.is;frame-ancestors 'self'",
    'content-type': 'application/json; charset=utf-8',
    date: 'Tue, 14 May 2024 13:03:57 GMT',
    etag: '"bwc9mymkdm2"',
    server: 'Vercel',
    'strict-transport-security': 'max-age=63072000',
    'x-dns-prefetch-control': 'on',
    'x-matched-path': '/api/send',
    'x-vercel-cache': 'MISS',
    'x-vercel-id': 'lhr1::iad1::h2z9t-1715691837425-0729f1aa741a'
  },
  body: ReadableStream { locked: false, state: 'readable', supportsBYOB: true },
  bodyUsed: false,
  ok: true,
  redirected: false,
  type: 'basic',
  url: 'https://track.tra.bz/api/send'
}
```

Any help would be much appreciated, as I can't log some stuff from the client side code...

Thanks,
Geoff

**Comment by boly38** [comment]
Hi
I created a gist here
https://gist.github.com/boly38/081318e620b6625467fef75bcb91b618

I'm able to produce some track event + page
but I encounter some issue while playing with nodeJs package + cloud instance 
(in some case: I didnt qualify exactly why/why not page track is working well on dashboard)

- maybe [doc](https://umami.is/docs/api/node-client) has no enough example
- no track of some page case
- confusion about user+session info depending on page hit or event hit (2 different sessions are created : one without props)
- how to link event with related page
- [npmjs package](https://www.npmjs.com/package/@umami/node) meta: git reference is broken // EDIT : repos is here : https://github.com/umami-software/node/ // Already fixed by umami-software/node#2

I'll maybe create dedicated issue later

**Reply by boly38** [reply]
I made a [v2 of the gist](https://gist.github.com/boly38/081318e620b6625467fef75bcb91b618/revisions) which work better : page hit + event hit (from a given page) from same user session +having props

**Reply by boly38** [reply]
created https://github.com/umami-software/node/issues/5
